patches/src/main/kotlin/app/revanced/patches/instagram/hide/reshare/HideReshareButtonPatch.kt

```diff
@@ -0,0 +1,80 @@
+package app.revanced.patches.instagram.hide.reshare
+
+import app.revanced.patcher.extensions.InstructionExtensions.addInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.getInstruction
+import app.revanced.patcher.patch.bytecodePatch
+import app.revanced.util.getReference
+import app.revanced.util.indexOfFirstInstructionOrThrow
+import com.android.tools.smali.dexlib2.Opcode
+import com.android.tools.smali.dexlib2.builder.instruction.BuilderSparseSwitchPayload
+import com.android.tools.smali.dexlib2.iface.instruction.OneRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.formats.Instruction31i
+import com.android.tools.smali.dexlib2.iface.reference.MethodReference
+
+@Suppress("unused")
+val hideReshareButtonPatch = bytecodePatch(
+    name = "Hide reshare button",
+    description = "Hides the reshare button from both posts and reels.",
+    use = false
+) {
+    compatibleWith("com.instagram.android")
+
+    execute {
+        feedResponseMediaParserFingerprint.method.apply {
+            // Each json field is parsed in a switch statement, where the case of the switch is the hashed field name.
+
+            /**
+             * First, find the switch payload where our field of interest is being processed. So find the payload that
+             * has a key == to our field of interest.
+             */
+            val switchPayload = implementation!!.instructions.first { ins ->
+                ins.opcode == Opcode.SPARSE_SWITCH_PAYLOAD &&
+                        (ins as BuilderSparseSwitchPayload).switchElements.any { it.key == hashedFieldInteger }
+            } as BuilderSparseSwitchPayload
+
+            // Get the target label, so find the instruction offset where the switch case is pointing to
+            val switchTargetLabel = switchPayload.switchElements
+                .first { it.key == hashedFieldInteger }
+                .target
+
+            // From that label, navigate forward until our field os interest is being instantiated
+            val moveResultIndex = indexOfFirstInstructionOrThrow(
+                switchTargetLabel.location.index,
+                Opcode.MOVE_RESULT_OBJECT
+            )
+
+            // Get its register
```

Comment:
```suggestion
            // Get its register.
```

Periods on the end of other comments too.